src/main/kotlin/io/hsar/wh40k/combatsimulator/model/World.kt

```diff
@@ -42,14 +43,26 @@ data class World(val friendlyForces: MutableList<UnitInstance>, val enemyForces:
                 .distanceToPosition(unitPositions.getValue(otherUnit))
     }
 
-    fun canMoveToUnit(unit: UnitInstance, otherUnit: UnitInstance, moveType: TurnAction.MoveAction): Boolean {
+    fun canMoveToUnit(unit: UnitInstance, otherUnit: UnitInstance, moveType: ActionOption.MoveAction): Boolean {
         return (distanceApart(unit, otherUnit) - 1 <=
                 unit.unit.stats.baseStats.getValue(BaseStat.AGILITY).getBonus()
                         .let { bonus ->
                             moveType.getMovementRange(bonus)
                         })
                 && moveType.isValidMovementPath(unitPositions.getValue(unit), unitPositions.getValue(otherUnit))
     }
+
+    fun getAdversaries(unit: UnitInstance): List<UnitInstance> {
```

Comment:
I see what you were saying about putting an `allegiance` flag on units now. If calls to this method get out of hand, we might need that.